[BUG] Failed to set special attribute [kueche_esszimmer.color_s]

### ✔️ Expected Behaviour

It whould work

### 🐞 Actual Behaviour

It gives the following error: ```TimeoutError: Failed to set special attribute [kueche_esszimmer.color_s]```

### 📋 Steps to Reproduce

```
active: true
type: OFFER
title: XXX
description: XXX
category: 80l16387/86l16387
special_attributes:
  kueche_esszimmer.color_s: sonstige
  kueche_esszimmer.art_s: geschirr
  kueche_esszimmer.condition_s: ok
price: 7
price_type: NEGOTIABLE
shipping_type: NOT_APPLICABLE
shipping_costs:
shipping_options:
sell_directly:
images:
  - ad_yyy__img1.jpg
contact:
  zipcode: 'XXX'
  name: XXX
  street:
  phone:
republication_interval: 3
id: XXX
created_on: '2024-02-16T00:00:00'
updated_on:
```

### 📺 What browsers are you seeing the problem on? (if applicable)

_No response_

### 💻 What operating systems are you seeing the problem on? (if applicable)

_No response_

### 📃 Relevant log output (if applicable)

```
[INFO] Deleting ad 'YYY' if already present...
[INFO]  ... pausing for 1954 ms ...
[INFO] Publishing ad 'YYY'...
[INFO]  ... pausing for 2225 ms ...
[INFO]  ... pausing for 1690 ms ...
[INFO]  ... pausing for 1796 ms ...
[INFO]  ... pausing for 2484 ms ...
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
args ({'code': -32000, 'message': 'Invalid search result range'},) kwargs {}
args ({'code': -32000, 'message': "DOM agent hasn't been enabled"},) kwargs {}
[ERROR] TimeoutError: Failed to set special attribute [kueche_esszimmer.color_s]
[73099] Failed to execute script '__main__' due to unhandled exception!
successfully removed temp profile /home/XXX/.config/google-chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

Also with `[ERROR] TimeoutError: Failed to set special attribute [spielzeug.condition_s]`

What I recognized: while publishining, the category is not choosen. This causes this ERROR.
The Bot click on category and then clicks on "continue/weiter" without selecting a category.

--- a/src/kleinanzeigen_bot/__init__.py
+++ b/src/kleinanzeigen_bot/__init__.py
@@ -718,7 +718,7 @@ class KleinanzeigenBot(WebScrapingMixin):
                     pass  # nosec
 
                 try:
-                    await self.web_select(By.XPATH, f"//select[@id='{special_attribute_key}']", special_attribute_value)
+                    await self.web_select(By.XPATH, f"//select[contains(@id, '{special_attribute_key}')]", special_attribute_value)
                 except TimeoutError:
                     LOG.debug("Attribute field '%s' is not of kind dropdown, trying to input as plain text...", special_attribute_key)
                     try:
@@ -726,7 +726,7 @@ class KleinanzeigenBot(WebScrapingMixin):
                     except TimeoutError:
                         LOG.debug("Attribute field '%s' is not of kind plain text, trying to input as radio button...", special_attribute_key)
                         try:
-                            await self.web_click(By.XPATH, f"//*[@id='{special_attribute_key}']/option[@value='{special_attribute_value}']")
+                            await self.web_click(By.XPATH, f"//*[contains(@id, '{special_attribute_key}')]/option[@value='{special_attribute_value}']")
                         except TimeoutError as ex:
                             LOG.debug("Attribute field '%s' is not of kind radio button.", special_attribute_key)
                             raise TimeoutError(f"Failed to set special attribute [{special_attribute_key}]") from ex

--- a/src/kleinanzeigen_bot/__init__.py
+++ b/src/kleinanzeigen_bot/__init__.py
@@ -649,6 +649,35 @@ class KleinanzeigenBot(WebScrapingMixin):
 
         utils.save_dict(ad_file, ad_cfg_orig)
 
+    async def __set_condition(self, condition_value: str) -> None:
+        condition_mapping = {
+            "new_with_tag": "Neu mit Etikett",
+            "new": "Neu",
+            "like_new": "Sehr Gut",
+            "alright": "Gut",
+            "ok": "In Ordnung",
+        }
+        mapped_condition = condition_mapping.get(condition_value)
+
+        try:
+            # Open condition dialog
+            await self.web_click(By.CSS_SELECTOR, '[class*="ConditionSelector"] button')
+        except TimeoutError:
+            LOG.debug("Unable to open condition dialog and select condition [%s]", condition_value, exc_info = True)
+            return
+
+        try:
+            # Click radio button
+            await self.web_click(By.CSS_SELECTOR, f'.SingleSelectionItem--Main input[type=radio][data-testid="{mapped_condition}"]')
+        except TimeoutError:
+            LOG.debug("Unable to select condition [%s]", condition_value, exc_info = True)
+
+        try:
+            # Click continue button
+            await self.web_click(By.XPATH, '//*[contains(@class, "ModalDialog--Actions")]//button[.//*[text()[contains(.,"Bestätigen")]]]')
+        except TimeoutError as ex:
+            raise TimeoutError("Unable to close condition dialog") from ex
+
     async def __set_category(self, ad_file:str, ad_cfg: dict[str, Any]) -> None:
         # click on something to trigger automatic category detection
         await self.web_click(By.ID, "pstad-descrptn")
@@ -674,6 +703,11 @@ class KleinanzeigenBot(WebScrapingMixin):
         if ad_cfg["special_attributes"]:
             LOG.debug('Found %i special attributes', len(ad_cfg["special_attributes"]))
             for special_attribute_key, special_attribute_value in ad_cfg["special_attributes"].items():
+
+                if special_attribute_key == "condition_s":
+                    await self.__set_condition(special_attribute_value)
+                    continue
+
                 LOG.debug("Setting special attribute [%s] to [%s]...", special_attribute_key, special_attribute_value)
                 try:
                     # if the <select> element exists but is inside an invisible container, make the container visible

